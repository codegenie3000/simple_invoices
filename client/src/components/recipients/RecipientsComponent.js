import React, {Component} from 'react';
// import {connect} from 'react-redux';
// import {requestRecipients} from '../../actions';
import ButtonLink from '../../components/global/ButtonLink';
import globalStyles from '../../stylesheets/GlobalElements.module.css';

import NoRecipients from '../../components/recipients/NoRecipients';
import RecipientTable from '../../components/recipients/RecipientsTable';

class RecipientsComponent extends Component {
    // fetches recipients using mapDispatchToProps from Recipients container when RecipientsComponent is mounted
    componentDidMount() {
        this.props.fetchRecipients();
    }

    // this.props.recipients is passed to RecipientsComponent from mapStateToProps from RecipientsContainer
    renderRecipients() {
        if (this.props.recipients.length < 1) {
            return <NoRecipients/>
        } else {
            return <RecipientTable recipients={this.props.recipients}/>
        }
    }

    render() {
        return (
            <div className={globalStyles.containerWhiteRounded}>
                <div className={globalStyles.flexOneOuterLeft}>
                    <h1>Invoice Recipients</h1><ButtonLink label='Add new recipient' to='/signUp'/>
                </div>
                <div>{this.renderRecipients()}</div>
            </div>
        );
    }
}

export default RecipientsComponent;