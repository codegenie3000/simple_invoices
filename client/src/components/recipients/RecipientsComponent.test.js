import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter, Route } from 'react-router-dom';

import RecipientsComponent from './RecipientsComponent';

function renderRecipientsComponent(recipientsData) {
    return shallow(
        <BrowserRouter>
            <Route>
                <RecipientsComponent recipients={recipientsData}/>
            </Route>
        </BrowserRouter>
    );
}

describe('<RecipientsComponent />', () => {
    it('renders with empty recipient array', () => {
        const recipientsData = [];
        const component = renderRecipientsComponent(recipientsData);
        expect(component).toMatchSnapshot();
    });
    it('renders with an array of recipients', () => {
        const recipientData = [
            {
                name: 'Jimmy McNulty',
                balance: 1000.25,
                id: 'abc123'
            },{
                name: 'Lester Freamon',
                balance: 200.00,
                id: 'xyz789'
            }
        ];
        const component = renderRecipientsComponent(recipientData);
        expect(component).toMatchSnapshot();
    });
});