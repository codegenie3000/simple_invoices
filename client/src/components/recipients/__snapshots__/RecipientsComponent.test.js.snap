// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipientsComponent /> renders with an array of recipients 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route>
    <RecipientsComponent
      recipients={
        Array [
          Object {
            "balance": 1000.25,
            "id": "abc123",
            "name": "Jimmy McNulty",
          },
          Object {
            "balance": 200,
            "id": "xyz789",
            "name": "Lester Freamon",
          },
        ]
      }
    />
  </Route>
</Router>
`;

exports[`<RecipientsComponent /> renders with empty recipient array 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route>
    <RecipientsComponent
      recipients={Array []}
    />
  </Route>
</Router>
`;
